Building Tracy convenience target requires LLD

### What happened?

I don't know if non-lld linkers are supposed to be supported on linux, but I forgot to set `-DIREE_ENABLE_LLD=ON` on a fresh build and was unable to build the tracy related tools (see the reproducer section for more details).

The issue is that the `libIREETracyNFD.a` library happens too early in the link list AFAICT.

The attached patch fixes that but I don't know if we care.
[tentative_fix.patch](https://github.com/iree-org/iree/files/10508140/tentative_fix.patch)


### Steps to reproduce your issue

To reproduce:
```
cmake -GNinja -DIREE_BUILD_TRACY=ON -DCMAKE_BUILD_TYPE=RelWithDebInfo  -DIREE_ENABLE_ASSERTIONS=OFF  <iree_checkout>
ninja iree-tracy-profiler iree-tracy-capture iree-tracy-csvexport
```

Result:
```
[1/1] Linking CXX executable tracy/iree-tracy-profiler
FAILED: tracy/iree-tracy-profiler 
: && /usr/bin/c++ -O2 -g -DNDEBUG  tracy/CMakeFiles/IREETracyProfiler.dir/__/__/__/third_party/tracy/profiler/src/BackendGlfw.cpp.o tracy/CMakeFiles/IREETracyProfiler.dir/__/__/__/third_party/tracy/profiler/src/ConnectionHistory.cpp.o tracy/CMakeFiles/IREETracyProfiler.dir/__/__/__/third_party/tracy/profiler/src/Filters.cpp.o tracy/CMakeFiles/IREETracyProfiler.dir/__/__/__/third_party/tracy/profiler/src/Fonts.cpp.o tracy/CMakeFiles/IREETracyProfiler.dir/__/__/__/third_party/tracy/profiler/src/HttpRequest.cpp.o tracy/CMakeFiles/IREETracyProfiler.dir/__/__/__/third_party/tracy/profiler/src/ImGuiContext.cpp.o tracy/CMakeFiles/IREETracyProfiler.dir/__/__/__/third_party/tracy/profiler/src/IsElevated.cpp.o tracy/CMakeFiles/IREETracyProfiler.dir/__/__/__/third_party/tracy/profiler/src/ResolvService.cpp.o tracy/CMakeFiles/IREETracyProfiler.dir/__/__/__/third_party/tracy/profiler/src/RunQueue.cpp.o tracy/CMakeFiles/IREETracyProfiler.dir/__/__/__/third_party/tracy/profiler/src/WindowPosition.cpp.o tracy/CMakeFiles/IREETracyProfiler.dir/__/__/__/third_party/tracy/profiler/src/main.cpp.o tracy/CMakeFiles/IREETracyProfiler.dir/__/__/__/third_party/tracy/profiler/src/winmain.cpp.o tracy/CMakeFiles/IREETracyProfiler.dir/__/__/__/third_party/tracy/profiler/src/winmainArchDiscovery.cpp.o -o tracy/iree-tracy-profiler  -ltbb  -lzstd  -lcapstone  -ldl  -lfreetype  -lglfw  -lgtk-3  -lgdk-3  -lpangocairo-1.0  -lpango-1.0  -lharfbuzz  -latk-1.0  -lcairo-gobject  -lcairo  -lgdk_pixbuf-2.0  -lgio-2.0  -lgobject-2.0  -lglib-2.0  -ldbus-1  tracy/libIREETracyIMGUI.a  tracy/libIREETracyCommon.a  tracy/libIREETracyNFD.a  tracy/libIREETracyServer.a  -lfreetype  -lglfw  -lgtk-3  -lgdk-3  -lpangocairo-1.0  -lpango-1.0  -lharfbuzz  -latk-1.0  -lcairo-gobject  -lcairo  -lgdk_pixbuf-2.0  -lgio-2.0  -lgobject-2.0  -lglib-2.0  -ldbus-1  tracy/libIREETracyCommon.a  -ltbb  -lzstd  -lcapstone  -ldl && :
/usr/bin/ld: tracy/libIREETracyServer.a(TracyFileselector.cpp.o): in function `tracy::Fileselector::OpenFile(char const*, char const*, std::function<void (char const*)>)':
<iree_checkout>/third_party/tracy/server/TracyFileselector.cpp:80: undefined reference to `NFD_OpenDialogN'
/usr/bin/ld: tracy/libIREETracyServer.a(TracyFileselector.cpp.o): in function `OpenFileImpl':
<iree_checkout>/third_party/tracy/server/TracyFileselector.cpp:84: undefined reference to `NFD_FreePathN'
/usr/bin/ld: tracy/libIREETracyServer.a(TracyFileselector.cpp.o): in function `tracy::Fileselector::SaveFile(char const*, char const*, std::function<void (char const*)>)':
<iree_checkout>/third_party/tracy/server/TracyFileselector.cpp:101: undefined reference to `NFD_SaveDialogN'
/usr/bin/ld: tracy/libIREETracyServer.a(TracyFileselector.cpp.o): in function `SaveFileImpl':
<iree_checkout>/third_party/tracy/server/TracyFileselector.cpp:105: undefined reference to `NFD_FreePathN'
/usr/bin/ld: tracy/libIREETracyServer.a(TracyFileselector.cpp.o): in function `tracy::Fileselector::Init()':
<iree_checkout>/third_party/tracy/server/TracyFileselector.cpp:19: undefined reference to `NFD_Init'
/usr/bin/ld: tracy/libIREETracyServer.a(TracyFileselector.cpp.o): in function `tracy::Fileselector::Shutdown()':
<iree_checkout>/third_party/tracy/server/TracyFileselector.cpp:26: undefined reference to `NFD_Quit'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed
```

### What component(s) does this issue relate to?

Other

### Version information

_No response_

### Additional context

_No response_

--- a/docs/developers/developing_iree/profiling_with_tracy.md
+++ b/docs/developers/developing_iree/profiling_with_tracy.md
@@ -108,7 +108,7 @@ A CMake-based build system for Tracy is maintained as part of IREE. In your IREE
 desktop build directory, set the following CMake option:
 
 ```shell
-$ cmake -DIREE_BUILD_TRACY=ON .
+$ cmake -DIREE_BUILD_TRACY=ON -DIREE_ENABLE_LLD=ON .
 ```
 
 That enables building the Tracy server tools, `iree-tracy-profiler` and

